src/System.Security.Cryptography.Primitives/tests/CryptoStream.netcoreapp.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Security.Cryptography.Encryption.Tests.Asymmetric
+{
+    public static partial class CryptoStreamTests
+    {
+        [Fact]
+        public static async Task DisposeAsync_DataFlushedCorrectly()
+        {
+            const string Text = "hello";
+
+            var stream = new MemoryStream();
+            using (CryptoStream encryptStream = new CryptoStream(stream, new IdentityTransform(64, 64, true), CryptoStreamMode.Write))
+            {
+                Assert.Equal(0, stream.Position);
+
+                byte[] toWrite = Encoding.UTF8.GetBytes(Text);
+                encryptStream.Write(toWrite, 0, toWrite.Length);
+                Assert.False(encryptStream.HasFlushedFinalBlock);
+                Assert.Equal(0, stream.Position);
+
+                await encryptStream.DisposeAsync();
+                Assert.True(encryptStream.HasFlushedFinalBlock);
```

Comment:
Perhaps `[Theory]` this to manually call FlushFinalBlock before the DisposeAsync, to show that it handles the manual user intervention.

Reply:
Ok, will do.